pkg/apis/middleware/session.go

```diff
@@ -24,6 +24,9 @@ func CreateTokenToSessionFunc(verify VerifyFunc) TokenToSessionFunc {
 			Email             string `json:"email"`
 			Verified          *bool  `json:"email_verified"`
 			PreferredUsername string `json:"preferred_username"`
+                        // TODO:  It's unclear whether the claim from --oidc-groups-claim should be here.
```

Comment:
Your intuition was correct, that option only applies to the `OIDCProvider` not this generic OIDC IDToken verifier used in extra bearer token validations.